app/controllers/v0/sign_in_controller.rb

```diff
@@ -181,6 +188,49 @@ def login_redirect_url(login_code, client_state = nil)
       redirect_uri.to_s
     end
 
+    def handle_authorize_error(error, status: :bad_request, level: :error)
+      context = { type: params[:type], client_state: params[:state], code_challenge: params[:code_challenge],
+                  code_challenge_method: params[:code_challenge_method] }
+      log_message_to_sentry(error.message, level, context)
+      sign_in_logger.authorize_stats(:failure, ["context:#{params[:type]}", VERSION_TAG, "error:#{error.try(:code)}"])
+      render json: { errors: error }, status: status
+    end
+
+    def handle_callback_error(error, status: :bad_request, level: :error)
+      context = { type: params[:type], state: params[:state], code: params[:code] }
+      log_message_to_sentry(error.message, level, context)
+      sign_in_logger.callback_stats(:failure, ["context:#{params[:type]}", VERSION_TAG, "error:#{error.try(:code)}"])
+      render json: { errors: error }, status: status
+    end
+
+    def handle_token_error(error, status: :bad_request, level: :error)
+      context = { code: params[:code], code_verifier: params[:code_verifier], grant_type: params[:grant_type] }
+      log_message_to_sentry(error.message, level, context)
+      sign_in_logger.token_stats(:failure, [VERSION_TAG, "error:#{error.try(:code)}"])
+      render json: { errors: error }, status: status
+    end
+
+    def handle_refresh_error(error, status: :unauthorized, level: :error)
+      context = { refresh_token: params[:refresh_token], anti_csrf_token: params[:anti_csrf_token] }
+      log_message_to_sentry(error.message, level, context)
+      sign_in_logger.refresh_stats(:failure, [VERSION_TAG, "error:#{error.try(:code)}"])
+      render json: { errors: error }, status: status
+    end
+
+    def handle_revoke_error(error, status: :unauthorized, level: :error)
+      context = { refresh_token: params[:refresh_token], anti_csrf_token: params[:anti_csrf_token] }
+      log_message_to_sentry(error.message, level, context)
+      sign_in_logger.revoke_stats(:failure, [VERSION_TAG, "error:#{error.try(:code)}"])
+      render json: { errors: error }, status: status
+    end
+
+    def handle_introspect_error(error, status: :unauthorized, level: :error)
+      context = { access_token: request.authorization }
```

Comment:
I actually would suggest we don't pass the `access_token` here. The reasoning is now that `authenticate` is in the `authentication.rb` file, the fact that we've hit this error means that authentication went fine and something in the introspect route raised the error. So we're actually logging a most likely valid `access_token` here, something we prob shouldn't do.

That being said, since we can assume `authenticate` worked, it means we can use `@current_user`. Perhaps we can just add `context = { user_uuid: @current_user.uuid }` instead?